### PR TITLE
Ability to add weekends to insights calculations

### PR DESCRIPTION
**Pull request by 1egoman**
screen shot 2018-06-15 at 11 34 44 am

- Uses the Density UI switch
- Adds this ability to both the insights list and detail pages

### CODE REVIEW

**Comment by 1egoman** [comment]
Don't merge yet, still needs tests. But the core mechanics work.

File: src/components/insights-space-list/index.js [hunk 1]
```diff
@@ -504,7 +532,7 @@ export class InsightsSpaceList extends React.Component {
                     </td>
                     <td className="insights-space-list-item-utilization">
                       <PercentageBar
-                        percentage={spaceUtilizations[space.id]}
+                        percentage={spaceUtilizations[space.id] || 0}
```
Comment (guscost) [inline src/components/insights-space-list/index.js]: Is this going to break rendering the em-dash?

File: src/components/insights-space-list/index.js [hunk 2]
```diff
@@ -153,9 +156,13 @@ export class InsightsSpaceList extends React.Component {
...
         const groups = groupCountsByDay(counts, space.timeZone);
         const filteredGroups = groups.filter(group => {
-          // Filter out any days that aren't within monday-friday
-          const dayOfWeek = moment.utc(group.date, 'YYYY-MM-DD').tz(space.timeZone).isoWeekday();
-          return dayOfWeek !== 5 && dayOfWeek !== 6 // Remove saturday and sunday
+          if (this.state.includeWeekends) {
+            return true; // Include all days
+          } else {
+            // Filter out any days that aren't within monday-friday
+            const dayOfWeek = moment.utc(group.date, 'YYYY-MM-DD').tz(space.timeZone).isoWeekday();
```
Comment (guscost) [inline src/components/insights-space-list/index.js]: Isn't this going to offset all days of the week by -1? UTC midnight is 8PM the previous day EDT.

File: src/components/insights-space-list/index.js [hunk 3]
```diff
@@ -110,7 +113,7 @@ export class InsightsSpaceList extends React.Component {
       let startTime,
           endTime = moment.utc().subtract(1, 'week').endOf('week').subtract(1, 'day').format('YYYY-MM-DDTHH:mm:ss');
       if (this.state.dataDuration === DATA_DURATION_WEEK) {
-        startTime = moment.utc().subtract(1, 'week').startOf('week').add(1, 'day').startOf('day').format('YYYY-MM-DDTHH:mm:ss');
+        startTime = moment.utc().subtract(1, 'week').startOf('week').subtract(1, 'day').startOf('day').format('YYYY-MM-DDTHH:mm:ss');
```
Comment (guscost) [inline src/components/insights-space-list/index.js]: ???

File: src/components/insights-space-detail-utilization-card/index.js [hunk 4]
```diff
@@ -380,14 +389,38 @@ export default class InsightsSpaceDetailUtilizationCard extends React.Component
                   ({TIME_SEGMENTS[this.state.timeSegment].name})
                 </span>
               </span>
+              <span className="insights-space-detail-utilization-card-header-weekends">
+                <span>Include Weekends</span>
+                <Switch
+                  enabled={this.state.includeWeekends}
+                  disabled={this.state.includeWeekendsProcessing}
```
Comment (guscost) [inline src/components/insights-space-detail-utilization-card/index.js]: Guessing this prop locks out all UI interactions. It's kinda confusingly named w/ `enabled` right above it.

File: src/components/insights-space-list/index.js (hunk 1 above)
Comment (1egoman) [reply src/components/insights-space-list/index.js]: Nope, already verified that

File: src/components/insights-space-detail-utilization-card/index.js (hunk 4 above)
Comment (1egoman) [reply src/components/insights-space-detail-utilization-card/index.js]: Yep I was thinking about that, maybe the `enabled` prop should be renamed something else?

File: src/components/insights-space-list/index.js (hunk 3 above)
Comment (1egoman) [reply src/components/insights-space-list/index.js]: Returns the last full week of data (the current week won't work as it's not complete yet)

File: src/components/insights-space-detail-utilization-card/index.js (hunk 4 above)
Comment (guscost) [reply src/components/insights-space-detail-utilization-card/index.js]: Maybe `value`? Is it truthy/falsy?

File: src/components/insights-space-list/index.js (hunk 3 above)
Comment (guscost) [reply src/components/insights-space-list/index.js]: Ah ok

Comment (1egoman) [reply src/components/insights-space-list/index.js]: I'll add some documentation to clarify

File: src/components/insights-space-detail-utilization-card/index.js (hunk 4 above)
Comment (1egoman) [reply src/components/insights-space-detail-utilization-card/index.js]: Yea, I'll do `value`.

**Comment by 1egoman** [comment]
I think this is ready to be merged.

**Pull request merged**